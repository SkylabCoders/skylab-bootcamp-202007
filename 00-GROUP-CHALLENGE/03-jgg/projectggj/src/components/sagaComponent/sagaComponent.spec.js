import React from 'react';
import renderer from 'react-test-renderer';
import SagaComponent from './sagaComponent';
import { BrowserRouter } from 'react-router-dom';

jest.mock('../../stores/store');


describe('SagaComponent', () => {




    it('should load sagas', () => {


        const sagaComponentTree = renderer.create(
            <BrowserRouter>
                <SagaComponent />
            </BrowserRouter>
        );

        stores._setSagas([{
            "series": "Z",
            "characters": ["Frieza", "Chi-Chi", "Android18", "Gabriel", "Kid_Buu", "Raditz", "Android17", "Goten", "Vegeta", "Gohan", "Android16", "Fat_Majin_Buu", "Cell"],
            "image": "https://i.pinimg.com/originals/60/a9/61/60a96199afa8469b7c3c46810ed86816.png",
        },
        {
            "series": "DragonBall",
            "characters": ["Goku", "gaaaaaaaa", "Piccolo", "Krillin", "goku", "Bulma", "krilin"],
            "image": "https://upload.wikimedia.org/wikipedia/commons/f/f3/Dragon_Ball_anime_logo.png"
        }]);


        expect(sagaComponentTree).toMatchSnapshot();

    })
});

