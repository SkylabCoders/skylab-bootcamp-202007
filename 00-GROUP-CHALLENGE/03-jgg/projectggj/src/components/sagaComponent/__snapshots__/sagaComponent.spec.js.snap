// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SagaComponent should load sagas 1`] = `
<div
  className="card-holder flex-item flex-col centred"
/>
`;
